lib/features/auth/login/login_navigator.dart

```diff
@@ -1,20 +1,21 @@
 import 'package:flutter_demo/dependency_injection/app_component.dart';
 import 'package:flutter_demo/features/auth/login/login_initial_params.dart';
 import 'package:flutter_demo/features/auth/login/login_page.dart';
+import 'package:flutter_demo/navigation/alert_dialog_route.dart';
 import 'package:flutter_demo/navigation/app_navigator.dart';
+import 'package:flutter_demo/navigation/error_dialog_route.dart';
 import 'package:flutter_demo/navigation/no_routes.dart';
 
-class LoginNavigator with NoRoutes {
+class LoginNavigator with NoRoutes, ErrorDialogRoute, AlertDialogRoute {
   LoginNavigator(this.appNavigator);
 
   @override
   final AppNavigator appNavigator;
 }
 
-//ignore: unused-code
 mixin LoginRoute {
   Future<void> openLogin(LoginInitialParams initialParams) async {
-    return appNavigator.push(
+    return appNavigator.pushReplacement(
```

Comment:
Not sure I am 'allowed' to modify the navigator for this task, but I did it because navigating like this raises a bug where users can press the device's back on Android within the login page and be redirected back to the 'splash page' where they would get stuck. The splash should be replaced when navigating and not be kept in stack